### PR TITLE
Add color interpolation widget to docs

### PR DESCRIPTION
**Pull request by graszka22**
## Summary
Adds the following section and a widget to the docs:
Screenshot 2022-12-30 at 17 23 16

### CODE REVIEW

File: docs/docs/api/miscellaneous/interpolateColors.mdx [hunk 1]
```diff
@@ -28,15 +30,31 @@ Can be either `'RGB'` or `'HSV'`.
...
+### Options explanation
+
+- _gamma_ - Colors on web / mobile are expressed in sRGB colorspace which is gamma-corrected, that is non-linear.
+  Operations on colors in non-linear space like addition will give wrong results. For example the interpolated color
+  may appear darker than it should. That's why when iterpolating we convert sRGB to linear space first and then convert
+  the result back into non-linear sRGB space. Gamma correction is device-dependent but for most devices to convert
+  from non-linear to linear space raising components to the power of gamma=2.2 is a good approximation. If you'd like to
+  disable that you can always set gamma=1. A nice article on that if you'd like to know more: https://observablehq.com/@sebastien/srgb-rgb-gamma
+- _useCorrectedHSVInterpolation_ - Sometimes (for example when interpolating from yellow to purple) HSV interpolation goes
+  through many other hues. This option allows to reduce the number of hues in such cases by treating hsv hues like a circular spectrum and choosing
```
Comment (tomekzaw) [inline docs/docs/api/miscellaneous/interpolateColors.mdx:51]:
```suggestion
  through many other hues. This option allows to reduce the number of hues in such cases by treating HSV hues like a circular spectrum and choosing
```

**Review by tomekzaw — CHANGES_REQUESTED** [review]
I'd like to propose some UX improvements:

1. Let's use lime `#00ff00` and pink `#ffc0cb` as the initial colors. The motivation is that these colors perfectly illustrate the differences between each interpolation mode:
rgb



2. I'm a huge fan of these color progress bars that you came up with. However, currently, it's not possible to see all 4 "predefined" modes at once (RGB, RGB+gamma, HSV, HSV+corrected). I would like to suggest adding the following table directly below color pickers:

| Mode | Result |
|-|:-:|
| RGB without gamma correction | Zrzut ekranu 2023-01-3 o 10 24 46 |
| RGB with gamma = 2.2 (default) | Zrzut ekranu 2023-01-3 o 10 25 06 |
| HSV without correction | Zrzut ekranu 2023-01-3 o 10 25 28 |
| HSV with correction | Zrzut ekranu 2023-01-3 o 10 25 44 |

3. Instead of having two separate color animation boxes ("RGB colorspace" and "HSV colorspace", I would like to suggest having only one and adding a following form:
* Color space: selectbox with two options: RGB, HSV
* Gamma: numeric input (disabled if HSV is selected)
* Use HSV corrected interpolation: checkbox (disabled if RGB is selected)

4. Currently, when you press "Run animation" button for the second time, nothing happens. It would be great if the animation could go back to the original color when pressed again, and so on. You can use [this](https://github.com/software-mansion/react-native-reanimated/blob/63960981c7b2b613bf99cc56f4fb7ed810ad57cb/FabricExample/src/TransformExample.tsx#L29-L32) code snippet.

**Comment by tomekzaw** [comment]
Some final suggestions:
1. Move "Example" section directly after "Options explanation"
2. Rename "Interpolate:" section header to "Playground"
3. Rename "Animation:" section header to "Live demo" 
4. Change gamma numeric input step from 1 to 0.1, set min value to 0.1 and max value to 10
5. I've changed my mind - instead of disabling the input/checkbox (depending on color space), make it hidden (i.e. render it conditionally)

File: docs/docs/api/miscellaneous/interpolateColors.mdx [hunk 2]
```diff
@@ -28,15 +30,27 @@ Can be either `'RGB'` or `'HSV'`.
...
 
 ### Returns
 
 `interpolateColor` returns the color after interpolation from within the output range in `rgba(r, g, b, a)` format.
 
+### Options explanation
+
+- _gamma_ - Colors on web / mobile are expressed in sRGB colorspace which is gamma-corrected, that is non-linear.
+  Operations on colors in non-linear space like addition will give wrong results. For example the interpolated color
+  may appear darker than it should. That's why when iterpolating we convert sRGB to linear space first and then convert
```
Comment (tomekzaw) [inline docs/docs/api/miscellaneous/interpolateColors.mdx:46]:
```suggestion
  may appear darker than it should. That's why when interpolating we convert sRGB to linear space first and then convert
```

File: docs/docs/api/miscellaneous/color_widget/styles.module.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+.marginLeft {
+    margin-left: 12px;
+}
+
+.form {
+    display: flex;
+    flex-direction: column;
+    row-gap: 24px;
+    font-weight: 700;
+    font-size:medium;
```
Comment (tomekzaw) [inline docs/docs/api/miscellaneous/color_widget/styles.module.css:34]:
```suggestion
    font-size: medium;
```

**Review by piaskowyk — APPROVED** [review]

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**